src/compiler/binder.ts

```diff
@@ -1297,6 +1297,9 @@ namespace ts {
                     // as other properties in the object literal.  So we use SymbolFlags.PropertyExcludes
                     // so that it will conflict with any other object literal members with the same
                     // name.
+                    if (options.strictThis) {
+                        seenThisKeyword = true;
+                    }

```

Comment:
Just to recap your findings: On the Monaco project this ends up adding as much as 10% to the check time for an overall impact of up to 5%.